src/upf/README.md

```diff
@@ -0,0 +1,166 @@
+# Read UPF Utility
+
+This module contains functionality to read, and modify information
+from Unified Power Format (UPF) files. 
+
+## Commands
+
+```{note}
+- Parameters in square brackets `[-param param]` are optional.
+- Parameters without square brackets `-param2 param2` are required.
+```
+
+### Read UPF
+
+Sources the UPF file. 
+
+```tcl
+read_upf
+    [-file file]
+```
+
+#### Options
+
+| Switch Name | Description | 
+| ----- | ----- |
+| `-file` | `.upf` filename |
+
+### Create Power Domain
+
+```tcl
+create_power_domain
+    [-elements elements]
+    name 
+```
+
+#### Options
+
+| Switch Name | Description | 
+| ----- | ----- |
+| `-elements` | list of module paths that belong this this domain OR `*` for top domain |
+| `name` | domain name |
+
+### Create Logic Port
+
+```tcl
+create_logic_port
+    [-direction direction]
+    port_name
+```
+
+#### Options
+
+| Switch Name | Description | 
+| ----- | ----- |
+| `-direction` | direction of the port (`in`, `out`, `inout`) |
+| `port_name` | port name |
+
+### Create Power Switch
+
+```tcl
+create_power_switch
+    [-domain domain]
+    [-output_supply_port output_supply_port]
+    [-input_supply_port input_supply_port]
+    [-control_port control_port]
+    [-on_state on_state]
+    name
+```
+
+#### Options
+
+| Switch Name | Description | 
+| ----- | ----- |
+| `-domain` | power domain name |
+| `-output_supply_port` | output supply port of the switch |
+| `-input_supply_port` | input supply port of the switch |
+| `-control_port` | a control port on the switch |
+| `-on_state` | {`state_name` `input_supply_port` {`boolean_expression`}} |
+| `name` | power switch name |
```

Comment:
```suggestion
| `-domain` | Power domain name. |
| `-output_supply_port` | Output supply port of the switch. |
| `-input_supply_port` | Input supply port of the switch. |
| `-control_port` | Control port on the switch. |
| `-on_state` | One of {`state_name` `input_supply_port` {`boolean_expression`}}. |
| `name` | Power switch name. |
```

Reply:
@luarss please capitalize the first word and add a `.` at the end of all sentences.